crypto/bio/bio_lib.c

```diff
@@ -252,7 +252,11 @@ static int bio_read_intern(BIO *b, void *data, size_t dlen, size_t *readbytes)
 {
     int ret;
 
-    if ((b == NULL) || (b->method == NULL) || (b->method->bread == NULL)) {
+    if (b == NULL) {
+        ERR_raise(ERR_LIB_BIO, BIO_R_NULL_PARAMETER);
+        return -2;
```

Comment:
Please use `ERR_R_PASSED_NULL_PARAMETER`...  I really see no reason why we even have a separate BIO_R for exactly the same reason.
Also, please return -1 on this error, as -2 is "traditionally" used for unsupported methods.

Reply:
Good point.
When I did that change I was actually wondering why the BIO module (as well has some further ones)
have their own NULL pointer argument error (BTW, "argument" would be better than "parameter" here).
For consistency, I've replaced all `BIO_R_NULL_PARAMETER` by `ERR_R_PASSED_NULL_PARAMETER` (and purged the former).

Reply:
I've also changed the return code in the above case.
Interesting to learn conventions about negative function result codes.
Are these documented/listed explicitly somewhere?

Reply:
I've extended the list in #10725 accordingly: https://github.com/openssl/openssl/issues/10725#issuecomment-748820356

Reply:
Re documentation, the 1.1.1 `man BIO_read` says this in the RETURN VALUES section:

> All other functions return either the amount of data successfully read
> or written (if the return value is positive) or that no data was
> successfully read or written if the result is 0 or -1. If the return
> value is -2 then the operation is not implemented in the specific BIO
> type. ...

Reply:
> I've also changed the return code in the above case.

Why only this case?

Reply:
Ah, I did not pay attention that the same issue was at other places in the same file.
I've improve them in the same way, and doing so also simplified various other return expressions in `bio_lib.c`.